### ISSUE TITLE
[Bug]: dn were crashed by OOM continuously during statbility test on distributed mode

### ISSUE THREAD

**Issue by aressu1985** [matrixorigin/matrixone#18045]
### Is there an existing issue for the same bug?

- [X] I have checked the existing issues.

### Branch Name

main

### Commit ID

c2fa177

### Other Environment Information

```Markdown
- Hardware parameters:
3*CN: 16C 64G
1*DN: 16C 64G
3*LOG: 4C 16G
3*PROXY 3C 7G
- OS type:
- Others:
```

### Actual Behavior

dn were crashed by OOM continuously during statbility test on distributed mode
<img width="1406" alt="image" src="https://github.com/user-attachments/assets/db9299b6-4608-4366-9c5c-ef35c72148f5">

dashborad link:
https://shanghai.idc.matrixorigin.cn:30001/d/85a562078cdf77779eaa1add43ccec1e/kubernetes-compute-resources-namespace-pods?orgId=1&var-datasource=prometheus&var-cluster=&var-namespace=mo-c2fa177-202408102118&from=1723384904347&to=1723426772967

mo log link:
https://shanghai.idc.matrixorigin.cn:30001/explore?panes=%7B%22qDo%22:%7B%22datasource%22:%22loki%22,%22queries%22:%5B%7B%22refId%22:%22A%22,%22expr%22:%22%7Bnamespace%3D%5C%22mo-c2fa177-202408102118%5C%22%7D%20%7C%3D%20%60%60%22,%22queryType%22:%22range%22,%22datasource%22:%7B%22type%22:%22loki%22,%22uid%22:%22loki%22%7D,%22editorMode%22:%22builder%22%7D%5D,%22range%22:%7B%22from%22:%221723384646484%22,%22to%22:%221723427727636%22%7D%7D%7D&schemaVersion=1&orgId=1

dn heap:
[oom.pprof.zip](https://github.com/user-attachments/files/16576756/oom.pprof.zip)

### Steps to Reproduce

```Markdown
1. run a mo cluster with config in this issue
2. run tpch 10G loop test processes in one independant tenant
3. run tpcc 10 warehouse and 10 ternimals longrunnig test processes in one independant tenant, prepare mode
4. run sysbench mixed cases(insert/delete/update/select) longrunnig test processes with 75 terminals in one independant tenant,non-prepare mode
5. run another sysbench mixed cases(insert/delete/update/select) longrunnig test processe with  75 terminals in one independant tenant,non-prepare mode
```

**Comment by jiangxinmeng1** [comment]
OOM是因为太久没有做CKP。

有个object上的refcount没清空导致刷不下去，然后ckp卡住了。refcount没清空应该是appender没关，这个pr修掉了 https://github.com/matrixorigin/matrixone/pull/18037

**Comment by jiangxinmeng1** [comment]
fixed by #18037

**Comment by aressu1985** [comment]
fixed
